public/js/controllers/status_controller.js

```diff
@@ -84,13 +86,14 @@ export default class extends Controller {
         while (bar.firstChild) bar.removeChild(bar.firstChild)
         bar.innerHTML = buildProgressBar(v)
 
-        if (v.subtitle === 'sync complete') {
+        if (v.subtitle === 'sync complete' && isRedicting) {
+          isRedicting = false // block consecutive calls.
```

Comment:
Can this event handler be run multiple times concurrently?  I don't know the limits of JS, but this would be a data race in go.

Reply:
I have tested this possibility several time but `hasRedirected` variable set outside the controller implementation ensures that redirecting function only runs once.